### PR TITLE
Add documentation for the ReSim log format and ROS2 converters

### PR DESCRIPTION
**Pull request by mikebauer**
## Description of change
Add some documentation for the ReSim log format and ROS2 converters.

## Guide to reproduce test results.
Render locally and read through.

## Checklist:
- [x] I have self-reviewed this change.
- [x] I have tested this change.
- [ ] This change is covered by tests that are already landed, or in this PR.

### CODE REVIEW

File: docs/resim/ros2/index.md [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+import resim.ros2.resim_log_from_ros2_python as rlr2
+from resim.ros2 import RESIM_DIR
+
+# ...
+
+rlr2.resim_log_from_ros2(
+    str(RESIM_DIR / "ros2" / "default_converter_plugin.so"), # See plugin info below
+    input_log_path,
+    converted_log_path,
+```
```
Comment (knowlsie) [inline docs/resim/ros2/index.md:42]: (nit) I think an extra new line would help the syntax highlighter here!

File: docs/resim/ros2/index.md [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+## Custom Message Types & Converter Plugins
+
+Occasionally, it is necessary or just convenient to convert a custom message
+type to a serialized ReSim protobuf type so that its contents can be used with
+ReSim's open source metrics and analysis tools. To facilitate this the
```
Comment (knowlsie) [inline docs/resim/ros2/index.md:47]:
```suggestion
ReSim's open source metrics and analysis tools. To facilitate this, the
```

**Review by knowlsie — APPROVED** [review]
LGTM

File: docs/resim/ros2/index.md (hunk 1 above)
Comment (mikebauer) [reply docs/resim/ros2/index.md:42]: I'm kind of unable to get the syntax highlighter working under any circumstances. 😕 
I've been tweaking the mkdocs.yml in our docs repository a bit, and I can only get highlighting which does not work in dark mode (the text is unreadable). I'll leave this as is for now.

File: docs/resim/ros2/index.md (hunk 2 above)
Comment (mikebauer) [reply docs/resim/ros2/index.md:47]: Good catch. Added!

**Review by sharonlinto — APPROVED** [review]

**Pull request merged**